packages/@react-aria/checkbox/src/useCheckboxGroupItem.ts

```diff
@@ -39,5 +40,13 @@ export function useCheckboxGroupItem(props: AriaCheckboxGroupItemProps, state: C
       }
     }
   });
-  return useCheckbox({...props, name: state.name}, toggleState, inputRef);
+
+  let {inputProps} = useCheckbox({
+    ...props,
+    isReadOnly: props.isReadOnly || state.isReadOnly,
+    isDisabled: props.isDisabled || state.isDisabled,
+    name: props.name || checkboxGroupNames.get(state)
```

Comment:
This doesn't make sense for me unless I miss something. It seems important to keep the same name for all "checkbox group items" as this "groups" them when sending the containing form element.

Reply:
I was thinking it might be possible to have a single "group" client side, but maybe you submit to a server using separate names for each checkbox.

Reply:
from what i remember of form submission, there are some specials names that work with some servers? i think the name has to include `"[]"` on the end
if you don't include that then checkboxes with the same name will overwrite each other
but not everything recognizes that special name syntax, so probably better to have them be individually named and just 'look' like a group

Reply:
> I was thinking it might be possible to have a single "group" client side, but maybe you submit to a server using separate names for each checkbox.

How realistic this use case is? Shouldn't this really be considered bad practice?